docs: use v1beta1 API version (#553)
open-feature/open-feature-operator: docs/feature_flag_source.md

# Flag Source configuration

The injected sidecar is configured using the `FeatureFlagSource` custom resource definition. 
The `openfeature.dev/featureflagsource` annotation is used to assign Pods with their respective `FeatureFlagSource` custom resources.

A minimal example of a `FeatureFlagSource` is given below,

```yaml
apiVersion: core.openfeature.dev/v1beta1
kind: FeatureFlagSource
metadata:
  name: feature-flag-source
spec:
  sources:                        # flag sources for the injected flagd
    - source: flags/sample-flags  # FeatureFlag - namespace/name
      provider: kubernetes        # kubernetes flag source backed by FeatureFlag custom resource
  port: 8080                      # port of the flagd sidecar
```

## Feature flag sources

This section explains how to configure feature flag sources to injected flag sidecar.

> `FeatureFlagSource` support multiple flag sources. Sources are configured as a list and given below are supported sources and their configurations,

Reviewer: Is this a comma-separated list or something else?  Should that be specified?

> `FeatureFlagSource` support multiple flag sources. Sources are configured as a list and given below are supported sources and their configurations,

Reviewer: ```suggestion
`FeatureFlagSource` support multiple flag sources. Sources are configured as a list .
Supported sources and their configurations are listed below.
```

### kubernetes aka `FeatureFlag`

This is `FeatureFlag` custom resource backed flagd feature flag definition.

> Read more on the custom resource at the dedicated documentation of [FeatureFlag](./feature_flag.md)

Reviewer: ```suggestion
Read more about the custom resource at the dedicated documentation of [FeatureFlag](./feature_flag.md)
```

> To refer this custom resource in `FeatureFlagSource`, provider type `kubernetes` is used as below example,

Reviewer: ```suggestion
The following example of a `FeatureFlagSource`, resource references this custom resource, using provider type `kubernetes` as the `provider` type:
```

## Sidecar configurations

> `FeatureFlagSource` further allows to provide configurations to the injected flagd sidecar.

Reviewer: ```suggestion
`FeatureFlagSource` can also provide configurations to the injected flagd sidecar.
```

        namespace: test-ns
        annotations:
            openfeature.dev/enabled: "true"
            openfeature.dev/featureflagsource: "config-A, test-ns-2/config-B"
```

In this example, 2 CRs are being used to configure the injected container (by default the operator uses the `flagd:main` image), `config-A` (which is assumed to be in the namespace `test-ns`) and `config-B` from the `test-ns-2` namespace, with `config-B` taking precedence in the configuration merge.

> The `FeatureFlagSource` version `v1beta1` CRD defines a CR with the following example structure, the documentation for this CRD can be found [here](crds.md#featureflagsource):

Reviewer: ```suggestion
The `FeatureFlagSource` version `v1beta1` CRD defines a CR with the following example structure.
The documentation for this CRD can be found
[here](crds.md#featureflagsource):
```

        memory: 256Mi
```

> The relevant `FeatureFlagSources` are passed to the operator by setting the `openfeature.dev/featureflagsource` annotation, and is responsible for providing the full configuration of the injected sidecar.

Reviewer: ```suggestion
The relevant `FeatureFlagSources` are passed to the operator by setting the `openfeature.dev/featureflagsource` annotation, which provides the full configuration of the injected sidecar.
```
Did I interpret this correction?  Is it the annotation what provides the full configuration?  "is responsible for providing" is unnecessarily verbose -- cleaner to just say "it provides"
